Add tests.yml

gotanygrapes-lemonade opened the pull request:
- add tests.yml
- update README badge pending

Commit:
Add tests.yml
- add tests.yml
- update README badge pending

Commit:
add workflows directory

Commit:
updated badge in README

gotanygrapes-lemonade committed:
Update tests.yml

fix the name of lint id to lambda-deploy

yang-l commented:
There are few tests failed, and need to be corrected before merging

Commit:
remove region in sample yml in README

Commit:
Merge branch 'migrate_CI_to_GitHubActions' of https://github.com/envato/lambda-deploy-buildkite-plugin into migrate_CI_to_GitHubActions

Commit:
add set -x to command

Commit:
specifying working-directory

Commit:
Tidy up the code:

- remove .buildkite directory
- run tests on pull_request
- remove "set -x" in command

orien commented on README.md:
```diff
@@ -1,6 +1,6 @@
-# Lambda Deploy Buildkite Plugin
+# Lambda Deploy GitHub Actions
```
This title doesn't look right.

orien commented on README.md:
```diff
@@ -1,6 +1,6 @@
-# Lambda Deploy Buildkite Plugin
+# Lambda Deploy GitHub Actions
 
-A [Buildkite plugin](https://buildkite.com/docs/agent/v3/plugins) for deploying AWS Lambda function code.
```
Let's keep this introductory sentence.

orien commented on README.md:
```diff
@@ -14,7 +14,6 @@ steps:
           path: path/to/deploy/code
           s3_bucket: deploybucket
           s3_key: deploy/key/name
-          region: ap-southeast-2
```
It looks like the code uses `region`. I don't think we should remove this from the example.

If the linter is complaining, we'll need to add `region` as a parameter in the `plugin.yml` file.

orien reviewed (commented):
Nice work.

Commit:
Adopt Orien's suggestions:

- change the title back to Lambda Deploy Buildkite Plugin
- keep buildkite introduction in README
- Add region back to README sample yml, add region to plugin.yml
